test/jdk/java/security/PEM/PEMDecoderTest.java

```diff
@@ -484,6 +484,19 @@ static void testClass(PEMData.Entry entry, Class clazz, boolean pass)
         }
     }
 
+    static void testClass(PEMData.Entry entry, Class clazz, boolean pass,
+        Class ec) throws RuntimeException {
+        try {
+            testClass(entry, clazz);
+        } catch (Exception e) {
+            if (e.getClass().isAssignableFrom(ec)) {
```

Comment:
You got it the other way around, testing if `e` is of a superclass of `ec`.
```suggestion
            if (ec.isInstance(e)) {
```
Or `ec.isAssignableFrom(e.getClass())`

Reply:
I cannot say the name is confusing because I actually don't know what it means. Over the years I've learnt to read it as `isParentOf`.

Reply:
`ec.isInstance(e)` make sense here